### PR TITLE
[Experimental] Product filters/remove template parts

### LINKED ISSUE #51780: Product Filters: Remove template parts
As the design direction changed on how we do the overlay, we need to remove the current template parts and their references:

- [x] Remove the Product Filters template part.
- [x] Remove the Product Filters Overlay template part.
- [x] Make the Product Filters block available from the inserter again.
- [x] Remove the Edit Overlay settings from the Product Filters block.

### PR DESCRIPTION
**Pull request by roykho**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/51780

This PR removes the template parts for the Product Filters and Product Filters Overlay. In addition to this:

- Remove overlay settings from Product Filters block
- Make Product Filters block insertable via the inserter

Note the reimplementation of the overlay and overlay navigation settings will be handled in this issue: https://github.com/woocommerce/woocommerce/issues/51781

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]
> **Test by developers only**
>
> **Pre-req:**
> - Ensure you have the WC Beta Tester Plugin installed and go to Tools > WCA Test Helper > Features. Toggle on the experimental-blocks option.
>  - Reset Product Catalog Template.

1. Go to Appearance > Editor > Patterns > General
2. Ensure you no longer see Product Filters Template Part and Product Filters Overlay Template Part
3. Go to Appearance > Editor > Templates > WooCommerce > Product Catalog
4. Add the Product Filters block using the inserter
5. Select Product Filters (Experimental) in the List View and go to the Inspector Controls > Styles tab
6. Ensure you no longer see any Overlay controls
7. Save the template and go to frontend Shop page
8. Ensure you see the Product Filters block displayed

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Experimental block - remove template parts
</details>

### CODE REVIEW

File: plugins/woocommerce/src/Blocks/Templates/ProductFiltersTemplate.php
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/Templates/ProductFiltersTemplate.php:1]: The class for Overlay should be removed too, right?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/overlay-navigation/block.json
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/overlay-navigation/block.json:1]: My observation on new design is: we don't allow users to customize the navigation button that much except choosing the style and icon, similar to Core Navigation block. So we won't have this block any more. It's best to confirm with Jill on this but whatever direction is correct, we can touch this block in #51781 so nothing more needs to be updated here, except the change you already made.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx [hunk 1]
```diff
@@ -1,42 +1,23 @@
...
 import { type BlockAttributes } from './types';
-import { BlockOverlayAttribute } from './constants';
 
 const TEMPLATE: InnerBlockTemplate[] = [
+	[
+		'woocommerce/product-filters-overlay-navigation',
+		{
+			triggerType: 'open-overlay',
+		},
+	],
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx]: I think we shouldn't add this this block to template in this PR. The button is visible always and clicking on it hiding itself.

**Review by dinhtungdu — DISMISSED** [review]
Thanks for working on this! I tested this PR locally and I can confirm the Product Filters blocks is now visible in the inserter and filter template parts are gone. I left some minor comments, otherwise, this is good to go.

**Comment by roykho** [comment]
@dinhtungdu - Thanks for the review. I have addressed the comments and pushed up updates.

**Review by thealexandrelara — APPROVED** [review]
Great work! I just reviewed the PR and everything seems good with the code. Here are the tests results:

# Test summary

✅ Scenario 1: Template parts are removed from the template parts list
- Conditions:
    - A merchant visits the template parts list (Appearance > Editor > Patterns> All Template Parts
- Expected Outcome: The "Product Filters (Experimental)" and "Filters Overlay (Experimental)" template parts are not available.
- Details: When a merchant visits the template part list, both the "Product Filters (Experimental)" and "Filters Overlay (Experimental)" template parts are not in the template parts list.

---

✅ Scenario 2:  The Product Filters block is available in the inserter
- Conditions:
    - A merchant visits a template.
    - The merchant opens up the Block Inserter and type "Product Filters".
- Expected Outcome: The "Product Filters (Experimental)" block should be in the list.
- Details: The Product Filters (Experimental) block is available in the block inserter.

---

✅ Scenario 3:  The Overlay controls are no longer available for the Product Filters block.
- Conditions:
    - A merchant visits a template.
    - The merchant adds the Product Filters block to the template.
    - The merchant selects the Product Filters block and check the Inspector Controls.
- Expected Outcome: No controls related to the Overlay are available.
- Details: The Overlay settings in the Inspector Controls are no longer available for the block.

---

✅ Scenario 1: When the store has no products, it correctly displays the Active and Attribute filter blocks for the Product Filters template part.
- Conditions:
    - The store has no products.
    - Merchant visits the Product Filters (Experimental) template part.
- Expected Outcome: The Product Filters (Experimental) block renders and correctly displays the Active and Attribute filter blocks.
- Details: The template part correctly displays the blocks.

--- 

✅ Scenario 4: When the store has no products, it correctly displays the Active and Attribute filter blocks for the Product Filters block.
- Conditions:
    - The store has no products.
    - Merchant visits the Product Catalog template.
    - Merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters  (Experimental) block renders and correctly displays the Active and Attribute filter blocks.
- Details: The template correctly displays the blocks.

---

✅  Scenario 5: When the store has products, it correctly displays the Active and Attribute filter blocks for the Product Filters block.
- Conditions:
    - The store has registered products.
    - Merchant visits the Product Catalog template.
    - Merchant adds the Product Filters (Experimental) block to the template.
- Expected Outcome: The Product Filters  (Experimental) block renders and correctly displays the Active and Attribute filter blocks.
- Details: The template correctly displays the blocks.

**Pull request merged**